subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/GradleVersionsPluginSmokeTest.groovy

```diff
@@ -58,14 +58,26 @@ class GradleVersionsPluginSmokeTest extends AbstractPluginValidatingSmokeTest {
             .forwardOutput()
 
-        def declarationConfiguration = ["compileClasspathCopy", "compileClasspathCopy2", "runtimeClasspathCopy", "runtimeClasspathCopy2", "testCompileClasspathCopy", "testCompileClasspathCopy2", "testRuntimeClasspathCopy", "testRuntimeClasspathCopy2"]
-        declarationConfiguration.each {
+        def nonDelcarableConfigurations = ["compileClasspathCopy", "compileClasspathCopy2", "runtimeClasspathCopy", "runtimeClasspathCopy2", "testCompileClasspathCopy", "testCompileClasspathCopy2", "testRuntimeClasspathCopy", "testRuntimeClasspathCopy2"]
+        nonDelcarableConfigurations.each { conf ->
             runner.expectDeprecationWarning(
-                "The $it configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 9.0. Please use another configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations",
-                "https://github.com/ben-manes/gradle-versions-plugin/issues/718"
+                    "The $conf configuration has been deprecated for dependency declaration. This will fail with an error in Gradle 9.0. Please use another configuration instead. Consult the upgrading guide for further information: https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_5.html#dependencies_should_no_longer_be_declared_using_the_compile_and_runtime_configurations",
+                    "https://github.com/ben-manes/gradle-versions-plugin/issues/718"
             )
         }
 
+        def copiedConfigurations = ["annotationProcessor", "compileClasspath", "runtimeClasspath", "testAnnotationProcessor", "testCompileClasspath", "testRuntimeClasspath", "classpath"]
+        copiedConfigurations.each { conf ->
+            ['sub1', 'sub2'].each { sub ->
+                runner.expectDeprecationWarning(
+                        "Copying configurations has been deprecated. This is scheduled to be removed in Gradle 9.0. Consider creating a new configuration and extending configuration ':$sub:$conf' instead. Consult the upgrading guide for further information: https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_8.html#configuration_copying_deprecated",
+                        "https://github.com/ben-manes/gradle-versions-plugin/issues/718")
+            }
+        }
+        runner.expectDeprecationWarning(
+                "Copying configurations has been deprecated. This is scheduled to be removed in Gradle 9.0. Consider creating a new configuration and extending configuration ':classpath' instead. Consult the upgrading guide for further information: https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_8.html#configuration_copying_deprecated",
```

Comment:
What's our recommendation for their use case here?

Reply:
Do you mean for `classpath` specifically?  Will the extension strategy not work?

Reply:
@big-guy - I added some documentation with an example alternative, and a sample test showing an extension made equal to the result of a copy.  The public API is limited in some things, see the test.  Would something like this be enough to go ahead with this?  I can't see why any affected plugins couldn't transition to using (more verbose) code like this to achieve whatever they were relying on `copy` to do.

Should the different pieces of copy (artifacts, attributes, etc.) be added to a public duplication helper API?

Reply:
I think manually copying things is still problematic in the same ways as `copy` is.  It's only a point-in-time copy.

I wasn't specifically thinking about `classpath`, but more the general issue and recommendation we're making here.

I think it's OK that we're saying you shouldn't use `copy`, but if we ask the versions plugin to use `extendsFrom`, I fear we're setting them up for another change in behavior. Haven't we talked about limiting which configurations can be "extended from"?  

What breaks if we also copy the intended role into the copy?

Reply:
Discuses last week, working on changes.